### PR TITLE
fix basic auth with custom user claim

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
We've fixed authentication with basic if oCIS is configured to use a non-standard claim
as user claim (`PROXY_USER_OIDC_CLAIM`). Prior to this bugfix the authentication always
failed and is now working.

## Related Issue

- Unblocks https://github.com/owncloud/ocis/issues/2387

## Motivation and Context
Make basic auth login work.

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by C0rby — APPROVED** [review]

File: proxy/pkg/middleware/basic_auth.go [hunk 1]
```diff
@@ -85,6 +85,7 @@ func BasicAuth(optionSetters ...Option) func(next http.Handler) http.Handler {
 				// fake oidc claims
 				claims := map[string]interface{}{
 					oidc.OwncloudUUID:      user.Id.OpaqueId,
+					options.UserOIDCClaim:  user.Id.OpaqueId,
```
Comment (rhafer) [inline proxy/pkg/middleware/basic_auth.go:88]: Doesn't this give you a dupliacte key here? E.g. when `options.UserOIDCClaim == oidc.Email` (which AFAIU even is the default)

**Review by rhafer — CHANGES_REQUESTED** [review]

File: proxy/pkg/middleware/basic_auth.go (hunk 1 above)
Comment (wkloucek) [reply proxy/pkg/middleware/basic_auth.go]: It indeed duplicates the key in some cases... But Go is fine with duplicate key as long as one of the duplicate keys is a variable... (https://play.golang.org/p/juVyt9SRnUf).

While looking at this I noticed, that `options.UserOIDCClaim` should only be set if use the `userid` to search for users on the CS3apis.

Comment (rhafer) [reply proxy/pkg/middleware/basic_auth.go]: Ah, thanks for clarifying. Didn't know that.

File: proxy/pkg/middleware/basic_auth.go [hunk 2]
```diff
@@ -84,10 +84,17 @@ func BasicAuth(optionSetters ...Option) func(next http.Handler) http.Handler {
...
 				claims := map[string]interface{}{
-					oidc.OwncloudUUID:      user.Id.OpaqueId,
 					oidc.Iss:               user.Id.Idp,
 					oidc.PreferredUsername: user.Username,
 					oidc.Email:             user.Mail,
+					oidc.OwncloudUUID:      user.Id.OpaqueId,
+				}
+
+				if options.UserCS3Claim == "userid" {
+					// set the custom user claim only if users will be looked up by the the userid on the CS3api
```
Comment (rhafer) [inline proxy/pkg/middleware/basic_auth.go:94]: Nit: double "the the"

**Review by rhafer — CHANGES_REQUESTED** [review]
Small nitpick about the comment. Otherwise lgtm.

**Review by rhafer — APPROVED** [review]

**Pull request merged**